src/api/v1/community_businesses/visit_logs.ts

```diff
@@ -60,7 +78,110 @@ const routes: Hapi.ServerRoute[] = [
       return CommunityBusinesses.addVisitLog(knex, activity, visitor);
     },
   },
-];
+  {
+    method: 'GET',
+    path: '/community-businesses/me/visit-logs',
+    options: {
+      description: 'Retrieve a list of visit logs for your community business',
+      auth: {
+        strategy: 'standard',
+        access: {
+          scope: ['visit_logs-own:read'],
+        },
+      },
+      validate: {
+        query: {
+          ...query,
+          ...filterQuery,
+        },
+      },
+      response: { schema: response },
+      pre: [
+        { method: getCommunityBusiness , assign: 'communityBusiness' },
+      ],
+    },
+    handler: async (request: GetVisitLogsRequest, h: Hapi.ResponseToolkit) => {
+      const {
+        server: { app: { knex } },
+        query: { limit, offset, filter: filterOptions = {} },
+        pre: { communityBusiness } } = request;
 
+      console.log({ filterOptions });
+
+      // move to query definition to model
+      const query = filter(valueIsSet, {
+        offset,
+        limit,
+        where: omit(['age'], { ...filterOptions }),
+        whereBetween: filterOptions.age
+        ? {
+          columnName: 'age',
+          range: filterOptions.age }
+        : {},
+      });
+
+      const visits = await CommunityBusinesses.getVisitLogs(
+        knex,
+        communityBusiness,
+        query
+        );
+
+      const count = await CommunityBusinesses.getVisitLogs(
+        knex,
+        communityBusiness,
+        omit(['limit', 'offset'], query)
+        ).then((rows: any) => rows.length);
+
+      return {
+        meta: {
+          total: count,
+        },
+        result: visits,
+      };
+    },
+  },
+  {
+    method: 'GET',
+    path: '/community-businesses/me/visit-logs/aggregates',
+    options: {
+      description: 'Retrieve a list of aggregated visit data for your scommunity businesses',
+      auth: {
+        strategy: 'standard',
+        access: {
+          scope: ['visit_logs-own:read'],
+        },
+      },
+      validate: {
+        query: {
+          ...query,
+          ...filterQuery,
+        },
+      },
+      response: { schema: response },
+      pre: [
+        { method: getCommunityBusiness , assign: 'communityBusiness' },
+      ],
+    },
+    handler: async (request: GetVisitLogsRequest, h: Hapi.ResponseToolkit) => {
+
+      const {
+        server: { app: { knex } },
+        query: { filter: filterOptions = {}, fields },
+        pre: { communityBusiness } } = request;
+
+      // deal with this in the model
+      const query = filter(valueIsSet, {
+        where: omit(['age'], { ...filterOptions }),
+        whereBetween: filterOptions.age
+          ? {
+            columnName: 'birthYear',
+            range: filterOptions.age}
```

Comment:
This would involve converting the age ranges to birth years on the client. IMO I don't think this is necessary. The way applyQueryModifiers is working with whereBetween, we can redefine the columnName in the model to the relevant `table.columnName`. I was keen to call `columnName` something less database specific, let me know if you have any suggestions.